pkg/storage/legacysql/dualwrite/filedb.go

```diff
@@ -4,96 +4,44 @@ import (
 	"context"
 	"encoding/json"
 	"os"
-	"sync"
-
-	"k8s.io/apimachinery/pkg/runtime/schema"
 
 	"github.com/grafana/grafana-app-sdk/logging"
 )
 
-// Simple file implementation -- useful while testing and not yet sure about the SQL structure!
-// When a path exists, read/write it from disk; otherwise it is held in memory
-type fileDB struct {
-	path    string
-	changed int64
-	db      map[string]StorageStatus
-	mu      sync.RWMutex
-	logger  logging.Logger
-}
-
-// File implementation while testing -- values are saved in the data directory
-func newFileDB(path string) *fileDB {
-	return &fileDB{
-		db:     make(map[string]StorageStatus),
-		path:   path,
-		logger: logging.DefaultLogger.With("logger", "fileDB"),
+// This format was used in early G12 provisioning config.  It should be removed
+func migrateFileDBTo(fpath string, db statusStorage) {
```

Comment:
I think this function should return an error and fail initialization if the migration failed